packages/icon-button/src/FzIconButton.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+  <button
+    type="button"
+    class="fz__iconbtn relative"
+    :disabled="disabled"
+    :class="classes"> 
+      <fz-icon 
+        class="fz__iconbtn__icon"
+        :name="iconName"
+        :variant="iconVariant"
+        :size="mappedIconSize"/>
+      <div class="fz__iconbtn__notification" 
+        v-if="variant === IconButtonVariant.notification"
+        :class="notificationClasses">
+      </div>
+      <span class="hidden w-0 h-0">{{ tooltip }}</span>
+    </button>
+</template>
+
+<script lang="ts" setup>
+import {computed} from 'vue'
+import {IconButtonVariant} from './types';
+import {FzIcon } from '@fiscozen/icons';
+import type { IconVariants } from '@fiscozen/icons';
+
+const props = withDefaults(
+  defineProps<{
+    /**
+     * The tooltip of the button will be shown on hover
+     */
+    tooltip: string
+    /**
+     * primary or secondary button
+     */
+    variant?: IconButtonVariant
+    /**
+     * size of the button
+     */
+    size?: 'sm' | 'md' | 'lg',
+    /**
+     * whether action is enabled or not
+     */
+    disabled: boolean,
+    /**
+     * Icon to be displayed. Use fontawesome search here https://fontawesome.com/v6/icons
+     * Mind that not all of the icons and variants might be available.
+     */
+    iconName: string,
+    /**
+     * Fontawesome icon variant: solid, regular, light, thin. Sharp subvariants are available as well
+     */
+    iconVariant?: IconVariants,
+  }>(),
+  { 
+    variant: IconButtonVariant.primary,
+    size: 'md',
+    disabled: false,
+    iconVariant: 'fasl',
+  }
+)
+
+const customVariantClasses = computed(() => ({
+  [IconButtonVariant.primary]: {
+    'bg-blue-500': true,
+    'hover:bg-blue-600': true,
+    'disabled:bg-blue-200': true, 
+    'text-core-white': true, 
+    'focus:bg-blue-500': !props.disabled,
+  },
+  [IconButtonVariant.secondary]: {
+    'text-grey-500': true, 
+    'bg-core-white': true,
+    'border-1': true,
+    'border-grey-200': true,
+    'hover:bg-grey-100': !props.disabled, 
+    'focus:border-blue-600': !props.disabled,
+    'disabled:text-grey-100': true,
+  },
+  [IconButtonVariant.notification]: {
+    'text-grey-500': true, 
+    'bg-core-white': true,
+    'border-1': true,
+    'border-grey-200': true,
+    'hover:bg-grey-100': !props.disabled, 
+    'disabled:text-grey-100': true,
+  },
+  [IconButtonVariant.invisible]: {
+    'text-grey-500': true, 
+    'bg-core-white': true,
+    'border-grey-200': true,
+    'hover:bg-grey-100': !props.disabled, 
+    'disabled:text-grey-100': true,
+  },
```

Comment:
l'unico problema che vedo con questo computed è che la variante può essere solo una, quindi elabora sempre tre oggetti inutilmente.

Reply:
potrei estrarre le parti statiche da quelle che dipendono da props.disabled, ma onestamente non credo si guadagni davvero... ho scelto di tenerlo così per semplicità di mantenimento: é più evidente "ad occhio" cosa viene applicato per le varianti

Reply:
per me è più rilevante evitare la creazione di tutta sta roba, potresti fare un switch case in modo che a runtime crei solo l'oggetto che ti serve in base alla variante